feat: bigdecimal support

hdbg opened the pull request:
Another take on bigdecimal support.

This is mainly for the databases: to avoid creating models, where just the `BigDecimal` is changed to `rust_decimal::Decimal`

hdbg committed:
feat: bigdecimal support

hdbg committed:
fix: added new bigdecimal features to the main crate

Drifter-J commented:
This is related to https://github.com/juhaku/utoipa/issues/1205 and https://github.com/juhaku/utoipa/issues/1449. 

I feel uneasy about this MR as we add an extra dependency which contradicts a philosophy of maintainer (@juhaku) based on his recent comment in https://github.com/juhaku/utoipa/issues/1565#issuecomment-4695453268 

<img width="858" height="252" alt="Screenshot 2026-06-13 at 23 56 43" src="https://github.com/user-attachments/assets/cbbca420-7163-48d4-81a7-fb912c0c46de" />

I still believe `value_type` is sufficient but it's up to @juhaku to decide whether adding a new dependency is acceptable in this case (as 2 peeps requested)

juhaku commented:
Well there are already 3 different time libraries supported so why not a second big decimal one. Seems like it is used quite a lot. This does not add a a runtime dependency, only for development, which is okayish. 

Though note for future, it would be worth to reassess the current hard coded types implementation whether there is another way around this which is more flexible and would need less hard coding in utoipa side. Perhaps the `utoipa-config` could have a facelift and be leveraged here as well.

juhaku reviewed (changes requested):
For this to merge, we still need CHAGELOG.md entry and the docs need to be updated including README.md 🙂

Commit:
docs: document bigdecimal and bigdecimal_float features

Add CHANGELOG entries and feature documentation in README.md and crate
docs for the new bigdecimal and bigdecimal_float feature flags.

Drifter-J committed:
Merge branch 'master' into master

Drifter-J commented:
> For this to merge, we still need CHAGELOG.md entry and the docs need to be updated including README.md 🙂

I pushed the doc changes :) and FYI, the examples test is failing due to unrelated issue! @juhaku

juhaku reviewed (approved):
🏆

juhaku commented:
> > For this to merge, we still need CHAGELOG.md entry and the docs need to be updated including README.md 🙂
> 
> I pushed the doc changes :) and FYI, the examples test is failing due to unrelated issue! @juhaku

Yeah, I am aware of that, that is super weird as the local build works ( or I have some different set of dependencies) :thinking: Anyhow it is related to some time crate compatibility with actix-web crate. Need to investigate at some time.

Drifter-J commented:
> > > For this to merge, we still need CHAGELOG.md entry and the docs need to be updated including README.md 🙂
> 
> > 
> 
> > I pushed the doc changes :) and FYI, the examples test is failing due to unrelated issue! @juhaku
> 
> 
> 
> Yeah, I am aware of that, that is super weird as the local build works ( or I have some different set of dependencies) :thinking: Anyhow it is related to some time crate compatibility with actix-web crate. Need to investigate at some time.

I can have a look today if you are not looking at it any time soon.
